src/controllers/dynakube/dynakube_controller.go

```diff
@@ -176,14 +175,23 @@ func (controller *DynakubeController) reconcileIstio(dynakube *dynatracev1beta1.
 }
 
 func (controller *DynakubeController) reconcileDynaKube(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
+	dtcFactory := dynatraceclient.NewFactory(controller.client, controller.dtcBuildFunc)
+	dtc, err := dtcFactory.Create(ctx, dynakube)
 
-	dtcReconciler := NewDynatraceClientReconciler(controller.client, controller.dtcBuildFunc)
-	dtc, err := dtcReconciler.Reconcile(ctx, dynakube)
 	if err != nil {
+		controller.setConditionTokenError(dynakube, err)
+		setStatusError := controller.updateDynakubeStatus(ctx, dynakube)
+
+		if setStatusError != nil {
+			log.Info("could not set dynakube status", "error", err.Error())
+		}
+
```

Comment:
in the event of an error we don't get to the end of the reconcile so the status would never be set

Reply:
```
	if err != nil { 
		requeueAfter = errorUpdateInterval

		var serverErr dtclient.ServerError
		isServerError := errors.As(err, &serverErr)
		if isServerError && serverErr.Code == http.StatusTooManyRequests {
			// should we set the phase to error ?
			log.Info("request limit for Dynatrace API reached! Next reconcile in one minute")
			return reconcile.Result{RequeueAfter: requeueAfter}, nil  // this only happens in case of a special error
		}
		dynakube.Status.SetPhase(dynatracev1beta1.Error)
	} else {
		dynakube.Status.SetPhase(controller.determineDynaKubePhase(dynakube))
	}

	isStatusDifferent, err := kubeobjects.IsDifferent(oldStatus, dynakube.Status)
	if err != nil {
		log.Error(err, "failed to generate hash for the status section")
	}
	if isStatusDifferent {
		log.Info("status changed, updating DynaKube")
		requeueAfter = changesUpdateInterval
		if errClient := controller.updateDynakubeStatus(ctx, dynakube); errClient != nil { // update here
			return reconcile.Result{}, errors.WithMessagef(errClient, "failed to update DynaKube after failure, original error: %s", err)
		}
	}
```

Reply:
i just noticed `oldStatus` being mutated if the status of the dynakube is mutated because the conditions are stored by reference. I.e., the copy `oldStatus` contained the same reference to the conditions array as the real status conditions and therefore always looked as if the same statuses were the same.

Fixed it by using `oldStatus := *dynakube.Status.DeepCopy()` instead to create a real copy